game.cpp

```diff
@@ -1928,6 +1947,16 @@ void test_game() //!OCLINT tests may be many
   }
 #endif
 
+//#define FIX_ISSUE_464
+#ifdef FIX_ISSUE_464
+  {
+    // (464) A player's state can be accessed easily
+    const game g;
+    const player_state default_state = player_state::active;
+    assert(get_nth_player_state(g, 0) == default_state);
```

Comment:
I enjoy the attempt and I think we both see how brittle this test is: it depends on the game's default state.

I suggest to compare the results of this logistic (as in, getting data from A to B) function with the longer way that is also possible, e.g. something like:

```c++
assert(get_nth_player_state(g, 0) == g.get_players()[0].get_state()); 
```

In that way, the logistic function is tested to be that actual shortcut.

It may feel like code duplication if the test does the same as the logistic function, but it isn't, as functions change. This test will guarantee the logisistic function doing the right thing.

Reply:
Another way to remove the brittleness:

```c++
const player_state default_state = get_default_player_state();
```

Reply:
I suggest your pick none, one or both of my suggested solutions and create an Issue for it :+1: